### PR TITLE
When charging stations in site are public, throw error if site is made private

### PR DESCRIPTION
**Pull request by RohanAntony**
Check if an error is thrown when a site with public charging stations is set to private.

### CODE REVIEW

**Comment by RohanAntony** [comment]
Do I need to log the Error?

**Comment by LucasBrazi06** [comment]
> Do I need to log the Error?
Normally, when you throw the exception, it should be already logged.

File: src/server/rest/v1/service/SiteService.ts [hunk 1]
```diff
@@ -392,6 +393,19 @@ export default class SiteService {
     site.name = filteredRequest.name;
     site.companyID = filteredRequest.companyID;
     if (Utils.objectHasProperty(filteredRequest, 'public')) {
+      if (!filteredRequest.public) {
+        const publicChargingStations = (await ChargingStationStorage.getChargingStations(req.tenant,
+          { siteIDs: [site.id], public: true, withSiteArea: true }, Constants.DB_PARAMS_MAX_LIMIT)).result;
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/SiteService.ts:398]: Limit to one charging stations and project only the charging station's ID

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

**Comment by LucasBrazi06** [comment]
Like the other issue, create an issue + link your PR to it + paste the link in Asana.

**Comment by RohanAntony** [comment]
@LucasBrazi06 While throwing the AppError, why do we not see the message in the notification that pops up? It seems like a longer process if the user has to go to the logs and read the message.

**Comment by LucasBrazi06** [comment]
> @LucasBrazi06 While throwing the AppError, why do we not see the message in the notification that pops up? It seems like a longer process if the user has to go to the logs and read the message.

You have to catch the exception in the front-end and display an i18n message ;-) so I would expect a PR in the dashboard also.

**Comment by LucasBrazi06** [comment]
@RohanAntony  Still not attached to an issue, just did it.

File: src/server/rest/v1/service/SiteService.ts [hunk 2]
```diff
@@ -392,6 +393,20 @@ export default class SiteService {
     site.name = filteredRequest.name;
     site.companyID = filteredRequest.companyID;
     if (Utils.objectHasProperty(filteredRequest, 'public')) {
+      if (!filteredRequest.public) {
+        const publicChargingStation = await ChargingStationStorage.getPublicChargingStationInSite(req.tenant, {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/service/SiteService.ts:397]: No need to create a specific method in Storage for this you can call the method right away.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
You have also to check when you set a charging station public on a non-public site.

**Review by LucasBrazi06 — APPROVED** [review]
Ok, still charging station to handle

**Pull request merged**